### ISSUE TITLE
Support commit message template functions

### ISSUE THREAD

**Issue by Dev25** [fluxcd/image-automation-controller#160]
The [current implementation](https://github.com/fluxcd/image-automation-controller/blob/main/controllers/imageupdateautomation_controller.go#L276-L288) of a commit message template relies on Go's `text/template` with no extra functions, it would be great to be able to use something like the sprig library which provides a bunch of functions (what Helm uses) or alternatively implement a small amount of functions depending on user demand.

http://masterminds.github.io/sprig/

This then allows users to have more complex logic when generating commits e.g. the following only works when there is 1 deployment, if i have multiple deployments following the same image policy then i need to count the unique strings in `.Updated.Images` which is not possible with the default `text/template` functions.
```go
        {{ if eq (len .Updated.Images) 1 -}}
        [skip ci] Auto Release of {{ index .Updated.Images 0 }}
        {{ else -}}
        [skip ci] Auto Release
        {{ end }}
```

**Comment by souleb** [comment]
can work on that.

**Comment by squaremo** [comment]
This seems like a decent idea and a reasonable request :-) My reservations are:

 - if sprig drags in a lot of dependencies, that makes upkeep more difficult (e.g., more chance for CVEs to apply)
 - we don't really want any functions that get access to the network or OS; if those can be disabled, then :+1:.

Another specific, tricky case is functions that access a source of randomness; I would err on the side of not including them.

**Comment by squaremo** [comment]
Re accessing OS and randomness: https://pkg.go.dev/github.com/Masterminds/sprig/v3#HermeticTxtFuncMap

**Comment by stefanprodan** [comment]
Implemented in #223
